public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -340,7 +340,7 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
 
   // By implementing getTagKeys and getTagValues we add ad-hoc filters functionality
   // Used in public/app/features/variables/adhoc/picker/AdHocFilterKey.tsx::fetchFilterKeys
-  getTagKeys(options?: DataSourceGetTagKeysOptions) {
+  getTagKeys(options?: DataSourceGetTagKeysOptions<InfluxQuery>) {
```

Comment:
```suggestion
  getTagKeys(options?: DataSourceGetTagKeysOptions) {
```

Think the type will be automatically inferred.

Reply:
I'd prefer having `<InfluxQuery>`.